src/read.c

```diff
@@ -2466,22 +2468,16 @@ static void RecreateStackNams( Obj context )
 ExecStatus ReadEvalCommand(Obj context, Obj *evalResult, UInt *dualSemicolon)
 {
     volatile ExecStatus          type;
-    volatile Obj                 stackNams;
-    volatile UInt                readTop;
-    volatile UInt                readTilde;
     volatile Obj                 tilde;
-    volatile UInt                currLHSGVar;
     volatile Obj                 errorLVars;
     syJmp_buf           readJmpError;
 #ifdef HPCGAP
     int                 lockSP;
 #endif
 
-    struct ReaderState * volatile rs = ReaderState();
-
-    ScannerState scanner;
-    ScannerState * volatile s = &scanner;
-    memset(s, 0, sizeof(ScannerState));
+    ReaderState reader;
+    ReaderState * volatile rs = &reader;
+    memset(rs, 0, sizeof(ReaderState));
```

Comment:
Up to here, basically nothing happened besides the things I explicitly pointed out; 99% of the changes are change `ScannerState * s` -> `ReaderState * rs` and related changes.

Here now we allocate a `ReaderState` on the stack, and zero it.

Reply:
Could we do reader = {0} instead?

Reply:
I actually tried this first, but it caused compiler warnings on Travis. I couldn't reproduce those on macOS or on my Ubuntu, probably because my versions of GCC were too new.

Reply:
No problem. Stupid travis, but it's not worth adding code which will have warnings.